feat: add the option to disable workflow

nejcr opened the pull request:
Closes https://github.com/keephq/keep/issues/1739

## 📑 Description

Added `is_disabled` to the `workflow` table and introduced `disabled` in the YAML configuration. Updated documentation and UI; see screenshots 



## ✅ Checks
- [ x ] My pull request adheres to the code style of this project
- [ x ] My code requires changes to the documentation
- [ x ] I have updated the documentation as required
- [ x ] All the tests have passed

## ℹ Additional Information
![image](https://github.com/user-attachments/assets/200d8345-5728-4b16-b468-3175bfa1c1a7)
![image](https://github.com/user-attachments/assets/997a0f91-fca7-48c7-8100-ce634d07fe7e)

Also spent an hour or two figuring out how to run `keep` in PyCharm; I might open a separate PR to document the process."




 /claim https://github.com/keephq/keep/issues/1739

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/keephq/keep?pullRequest=1766) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/keephq/keep?pullRequest=1766) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/keephq/keep?pullRequest=1766) it.</sub>

talboren commented:
Not sure if I'm right but it might conflict with: https://github.com/keephq/keep/pull/1488 which is about to get merged.

nejcr commented:
> Not sure if I'm right but it might conflict with: #1488 which is about to get merged.

I can wait, looking at it, it would be easier for me if I resolved conflicts afterwards

rajeshj11 commented:
@nejcr I think your changes are conflicting with my changes. please resolve them. post-resolution can you also share the final changes with me? If needed any clarification on new changes please do reach out to me.

nejcr commented:
@shahargl I have resolved the conflicts with the main branch.

@rajeshj11 I incorporated some of your validator changes as they were necessary for proper functionality. The code should now be free of conflicts. Additionally, I simplified the nested ternary logic for better readability.

rajeshj11 commented on keep-ui/app/workflows/builder/builder-validators.tsx:
```diff
@@ -17,10 +17,11 @@ export function globalValidatorV2(
+    !definition.properties['manual'] &&
+    !definition.properties['interval'] &&
+    !definition.properties['alert'] &&
+    !definition.properties['disabled']
```
I think this check is not needed. this is not trigger related data.
!definition.properties['disabled']

nejcr commented:
Thank you, @Matvey-Kuk.

I have rebased the changes from the main branch to keep this PR up to date.

Please let me know if anything else is needed. (Note: There were some database setup issues with the tests running on GitHub Actions.)

Matvey-Kuk commented:
@nejcr I see the tests are still failing, please let me know if you are planning to fix them

Matvey-Kuk commented:
Great work, thank you!

nejcr commented:
@Matvey-Kuk @talboren another migration got into main before aproved workflows were triggered.
I've had to resolve another double headed migration.

nejcr commented:
@Matvey-Kuk looks like something changed so the now the pr tests are red.
I have added a possible fix(passes on 1739 branch).

Matvey-Kuk commented:
Thank you!
